app/controllers/repositories_controller.rb

```diff
@@ -0,0 +1,101 @@
+class RepositoriesController < ApplicationController
+  def index
+    @repositories = Repository.where(hide: false).paginate(:page => params[:page])
+    respond_to do |format|
+        format.html
+        format.js
+    end
+  end
+
+  def show
+    @repository = Repository.find(params[:id])
```

Comment:
Can we have a common method to find repo by id so that we can reuse across many actions in this controller?